mongo/options/clientencryptionoptions.go

```diff
@@ -29,6 +36,36 @@ func (c *ClientEncryptionOptions) SetKmsProviders(providers map[string]map[strin
 	return c
 }
 
+// SetTLSOptions specifies tls.Config instances for each KMS provider to use to configure TLS on all connections created
+// to the cluster. The input map should contain a mapping from each KMS provider to a document containing the necessary 
+// options, as follows:
+//
+// {
+//		"kmip": {
+//			"tlsCertificateKeyFile": "foo.pem",
+// 			"tlsCAFile": "fooCA.pem"
+//		}
+// }
+//
+// Currently, the following TLS options are supported:
+//
+// 1. "tlsCertificateKeyFile" (or "sslClientCertificateKeyFile"): The "tlsCertificateKeyFile" option specifies a path to 
+// the client certificate and private key, which must be concatenated into one file. 
+//
+// 2. "tlsCertificateKeyFilePassword" (or "sslClientCertificateKeyPassword"): Specify the password to decrypt the client
+// private key file (e.g. "tlsCertificateKeyFilePassword=password").
+//
+// 3. "tlsCaFile" (or "sslCertificateAuthorityFile"): Specify the path to a single or bundle of certificate authorities
+// to be considered trusted when making a TLS connection (e.g. "tlsCaFile=/path/to/caFile").
+//
+// This should only be used to set custom TLS options. By default, the connection will use an empty tls.Config{}.
+func (c *ClientEncryptionOptions) SetTLSOptions(tlsOpts map[string]map[string]interface{}) (*ClientEncryptionOptions, error) {
+	tlsConfig := make(map[string]*tls.Config)
+	tlsConfig, err := applyTLSOptions(tlsOpts, tlsConfig)
+	c.TLSConfig = tlsConfig
+	return c, err
+}
+
```

Comment:
We eventually want to add a `SetTLSConfig(cfg *tls.Config)` to both `ClientEncryptionOptions` and `AutoEncryptionOptions`  to support users who currently build their own TLS configs and set them via the `SetTLSConfig` function on a `ClientOptions`. We can accomplish that here or in a follow-up PR.

If a client sets their own TLS config, we should set `MinVersion: tls.VersionTLS12` on the passed-in config unless the user explicitly sets `MinVersion`.

Reply:
Was able to add a `SetTLSConfig(cfg map[string]*tls.Config)` function to `ClientEncryptionOptions` and `AutoEncryptionOptions` in the API refactor. Also set the min TLS version of tls.Configs passed in and added a comment explaining why the min version is set to 1.2.